content/docs/intro/concepts/programming-model.md

```diff
@@ -802,138 +1457,110 @@ record = aws.route53.Record("validation",
+This approach is easier to read and write, while not losing any important dependency information that is needed to properly create and maintain the stack. This approach doesn't work in all cases, but when it does, it can be a great help.
 
-{{< langchoose csharp >}}
+> In JavaScript and TypeScript, a 'lifted' property access on an `Output<T>` that wraps `undefined` will produce another `Output<T>` with the `undefined` value instead of throwing or producing a 'faulted' `Output<T>`. In this manner lifted property accesses behave like the [`?.` (optional chaining operator)](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#optional-chaining) in JavaScript and TypeScript. This makes it much easier to form a chain of property accesses on an `Output<T>`.
 
 ```javascript
-function split(input) {
-    let output = pulumi.output(input);
-    return output.apply(v => v.split());
-}
-```
+let certValidation = new aws.route53.Record("cert_validation", {
+  records: [certCertificate.domainValidationOptions[0].resourceRecordValue],
 
-```typescript
-function split(input: pulumi.Input<string>): pulumi.Output<string[]> {
-    let output = pulumi.output(input);
-    return output.apply(v => v.split());
-}
+// instead of
+
+let certValidation = new aws.route53.Record("cert_validation", {
+  records: [certCertificate.apply(cc => cc ? cc.domainValidationOptions : undefined)
+                           .apply(dvo => dvo ? dvo[0] : undefined)
+                           .apply(o => o ? o.resourceRecordValue : undefined)],
 ```
 
-```python
-def split(input):
-    output = Output.from_input(input);
-    return output.apply(lambda v: v.split());
-}
-```
+```typescript
+let certValidation = new aws.route53.Record("cert_validation", {
+  records: [certCertificate.domainValidationOptions[0].resourceRecordValue],
 
-```go
-// Helpers for converting Inputs to Outputs are not yet available in Go.
-//
-// See https://github.com/pulumi/pulumi/issues/1614.
-```
+// instead of
 
-```csharp
-Output<string[]> Split(Input<string[]> input)
-{
-    var output = input.ToOutput()
-    return output.Apply(v => v.Split(","));
-}
+let certValidation = new aws.route53.Record("cert_validation", {
+  records: [certCertificate.apply(cc => cc ? cc.domainValidationOptions : undefined)
+                           .apply(dvo => dvo ? dvo[0] : undefined)
+                           .apply(o => o ? o.resourceRecordValue : undefined)],
 ```
 
-##### Working with Outputs and strings {#outputs-and-strings}
+##### Working with Outputs and Strings {#outputs-and-strings}
+
+Outputs containing strings cannot be used directly in operations like string concatenation. _String interpolation_ lets you more easily build a string out of various output values, without needing {{< pulumi-apply >}} or {{< pulumi-all >}}. This can be used to export a stack output, provide a dynamically computed string as a new resource argument, or even just for diagnostics purposes.
```

Comment:
```suggestion
Outputs containing strings cannot be used directly in operations like string concatenation. _String interpolation_ lets you more easily build a string out of various output values, without needing {{< pulumi-apply >}} or {{< pulumi-all >}}. This can be used to export a stack output, provide a dynamically computed string as a new resource argument, or even just for diagnostic purposes.
```